app/controllers/api/api_controller.rb

```diff
@@ -8,7 +8,7 @@ class Api::ApiController < ApplicationController
    private
    def authenticated?
      authenticate_or_request_with_http_basic do |email, password|
-       User.where( email: email).first.valid_password?(password)
+       @current_user = User.find_by_email(email) if User.where( email: email).first.valid_password?(password)
```

Comment:
you are performing 2 queries on this line. User.where gets the user object you need and confirms the password. since it already grabbed the user, you could store it, check password and if that returns true, then load into current_user